numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -1990,14 +1990,8 @@ string_partition_strided_loop(
             goto fail;
         }
 
-        npy_intp idx;
-
-        if (startposition == STARTPOSITION::FRONT) {
-            idx = fastsearch((char *)i1s.buf, i1s.size, (char *)i2s.buf, i2s.size, -1, FAST_SEARCH);
-        }
-        else {
-            idx = fastsearch((char *)i1s.buf, i1s.size, (char *)i2s.buf, i2s.size, -1, FAST_RSEARCH);
-        }
+        int direction = startposition == STARTPOSITION::FRONT ? FAST_SEARCH : FAST_RSEARCH;
```

Comment:
Would it make sense to explicitly take this out of the loop? Or would the compiler optimize that anyway?  It probably is a bit clearer right next to the definition of `startposition`